server/tls/tls.go

```diff
@@ -64,12 +69,36 @@ func Config(certFile string, keyFile string) (*crypto_tls.Config, error) {
 		MinVersion:               config.Config.Server.TLS.Override.MinVersion,
 		MaxVersion:               config.Config.Server.TLS.Override.MaxVersion,
 		CipherSuites:             config.Config.Server.TLS.Override.CipherSuites,
-		Certificates:             []crypto_tls.Certificate{cert},
+		GetCertificate:           returnCert,
+	}
+
+	if len(certificates) == 0 {
+		certificates = make(map[string]*crypto_tls.Certificate)
+	}
+	certificates[domain] = &cert
+
+	// GetCertificate returns a Certificate based on the given
```

Comment:
simplify + add reference to original code:
https://golang.org/pkg/crypto/tls/#Config.GetCertificate